src/charm.py

```diff
@@ -94,6 +104,35 @@ def _pebble_layer(self) -> Dict:
             },
         }
 
+    def _on_reset_instance_action(self, event: ActionEvent) -> None:
```

Comment:
This action is pretty dangerous, shouldn't we add a parameter like "config" that needs to be set to "IUNDERSTANDITWILLDELETEALLMYDATA" (or something shorter 😄

Reply:
Well if you add the confirm parameter, that warns you, and you didn't want to execute this we can't do much for you.
But I still think it's a safeguard that avoids launching the action by mistake, but I agree that ideally it should ask for confirmation before execution but that's no possible in juju